cmd/registry/cmd/label.go

```diff
@@ -0,0 +1,260 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/apigee/registry/cmd/registry/core"
+	"github.com/apigee/registry/connection"
+	"github.com/apigee/registry/gapic"
+	"github.com/apigee/registry/rpc"
+	"github.com/apigee/registry/server/names"
+	"github.com/spf13/cobra"
+	"google.golang.org/genproto/protobuf/field_mask"
+)
+
+var labelFilter string
+var labelOverwrite bool
+var labelsToSet map[string]string
+var labelsToClear []string
+
+func init() {
+	rootCmd.AddCommand(labelCmd)
+	labelCmd.Flags().StringVar(&labelFilter, "filter", "", "Filter selected resources")
+	labelCmd.Flags().BoolVar(&labelOverwrite, "overwrite", false, "Overwrite existing labels")
+}
+
+var labelCmd = &cobra.Command{
+	Use:   "label RESOURCE KEY_1=VAL_1 ... KEY_N=VAL_N",
+	Short: "Label resources in the API Registry",
+	Args:  cobra.MinimumNArgs(2),
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := context.TODO()
+		client, err := connection.NewClient(ctx)
+		if err != nil {
+			log.Fatalf("%s", err.Error())
+		}
+
+		taskQueue := make(chan core.Task, 1024)
+		workerCount := 64
+		for i := 0; i < workerCount; i++ {
+			core.WaitGroup().Add(1)
+			go core.Worker(ctx, taskQueue)
+		}
+
+		labelsToClear = make([]string, 0)
+		labelsToSet = make(map[string]string)
```

Comment:
We aren't validating that these are disjoint sets. If a label is set and cleared in the same command it will arbitrarily have one behavior chosen.

Reply:
True that we aren't checking for disjoint sets. The behavior would be consistent though, always removing first and then setting, effectively overwriting an existing label.